### PR TITLE
Fix titanic example add_dataset_objective

### LINKED ISSUE #167: Permissions fields are not set in the examples
It was done automatically by the SDK. This behaviour has been changed in 34142669105be56f85e28718f304f4ba4894358b. That's the reason why this field should be set manually in the examples.

### PR DESCRIPTION
**Pull request by natct10**
Add permissions to ALGO, DATASET & OBJECTIVE
Fix #167

### CODE REVIEW

**Comment by natct10** [comment]
> Until now the default permissions were "private", I think they should stay that way. This will avoid unwarranted leaks if the code from the example is adapted for a real-life use case.

Okay, thank you for this input, I'll update!

**Comment by natct10** [comment]
In fact, I am still getting a 400 error:
```sh
Adding test data samples...
100%|███████████████████████████████████| 2/2 [00:00<00:00,  6.89it/s]
Associating data samples with dataset...
Adding objective...
Requests error status 400: {"permissions":["This field may not be null."]}
Traceback (most recent call last):
  File "/home/zen/.local/lib/python3.7/site-packages/substra/sdk/rest_client.py", line 126, in __request
    r.raise_for_status()
  File "/usr/lib/python3/dist-packages/requests/models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http://substra-backend.node-1.com/objective/

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "scripts/add_dataset_objective.py", line 144, in <module>
    }, exist_ok=True)['pkhash']
  File "/home/zen/.local/lib/python3.7/site-packages/substra/sdk/client.py", line 41, in wrapper
    return f(*args, **kwargs)
  File "/home/zen/.local/lib/python3.7/site-packages/substra/sdk/client.py", line 323, in add_objective
    res = self._add(assets.OBJECTIVE, data, files=files, exist_ok=exist_ok)
  File "/home/zen/.local/lib/python3.7/site-packages/substra/sdk/client.py", line 164, in _add
    **requests_kwargs)
  File "/home/zen/.local/lib/python3.7/site-packages/substra/sdk/rest_client.py", line 261, in add
    return self._add(name, exist_ok=exist_ok, **request_kwargs)
  File "/home/zen/.local/lib/python3.7/site-packages/substra/sdk/rest_client.py", line 237, in _add
    return self.request('post', name, **request_kwargs)
  File "/home/zen/.local/lib/python3.7/site-packages/substra/sdk/utils.py", line 170, in wrapper
    return f(*args, **kwargs)
  File "/home/zen/.local/lib/python3.7/site-packages/substra/sdk/rest_client.py", line 192, in request
    **request_kwargs,
  File "/home/zen/.local/lib/python3.7/site-packages/substra/sdk/rest_client.py", line 170, in _request
    return self.__request(request_name, url, **request_kwargs)
  File "/home/zen/.local/lib/python3.7/site-packages/substra/sdk/rest_client.py", line 138, in __request
    raise exceptions.InvalidRequest.from_request_exception(e)
substra.sdk.exceptions.InvalidRequest: 400 Client Error: Bad Request for url: http://substra-backend.node-1.com/objective/
```

File: examples/titanic/scripts/add_dataset_objective.py [hunk 1]
```diff
@@ -71,6 +75,10 @@ def progress_bar(length):
     'description': os.path.join(assets_directory, 'objective/description.md'),
     'metrics_name': 'accuracy',
     'metrics': os.path.join(assets_directory, 'objective/metrics.zip'),
+    "permissions": {
+        "public": False,
+        "authorized_ids": []
+    },
```
[the remark this answers is not in the document]
Comment (jmorel) [reply examples/titanic/scripts/add_dataset_objective.py:81]: But you need to add them to the objective.

[the remark this answers is not in the document]
Comment (jmorel) [reply examples/titanic/scripts/add_dataset_objective.py:81]: My bad, the diff made me thing the change was not the one expected.

**Comment by Fabien-GELUS** [comment]
I get the same error with the latest commit of this branch with the command : sudo python3 scripts/add_dataset_objective.py
```
Adding dataset...
Requests error status 400: {"message":[{"name":["This field may not be null."]}],"pkhash":"c1d9b42d9538f825109c38c4f599d47391347d198a0770331ba790b7ebcfaa40"}
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/substra/sdk/rest_client.py", line 126, in __request
    r.raise_for_status()
  File "/usr/lib/python3/dist-packages/requests/models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http://substra-backend.node-1.com/data_manager/

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "scripts/add_dataset_objective.py", line 95, in <module>
    dataset_key = client.add_dataset(DATASET, exist_ok=True)['pkhash']
  File "/usr/local/lib/python3.8/dist-packages/substra/sdk/client.py", line 41, in wrapper
    return f(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/substra/sdk/client.py", line 288, in add_dataset
    res = self._add(assets.DATASET, data, files=files, exist_ok=exist_ok)
  File "/usr/local/lib/python3.8/dist-packages/substra/sdk/client.py", line 160, in _add
    return self.client.add(
  File "/usr/local/lib/python3.8/dist-packages/substra/sdk/rest_client.py", line 261, in add
    return self._add(name, exist_ok=exist_ok, **request_kwargs)
  File "/usr/local/lib/python3.8/dist-packages/substra/sdk/rest_client.py", line 237, in _add
    return self.request('post', name, **request_kwargs)
  File "/usr/local/lib/python3.8/dist-packages/substra/sdk/utils.py", line 170, in wrapper
    return f(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/substra/sdk/rest_client.py", line 189, in request
    response = self._request(
  File "/usr/local/lib/python3.8/dist-packages/substra/sdk/rest_client.py", line 170, in _request
    return self.__request(request_name, url, **request_kwargs)
  File "/usr/local/lib/python3.8/dist-packages/substra/sdk/rest_client.py", line 138, in __request
    raise exceptions.InvalidRequest.from_request_exception(e)
substra.sdk.exceptions.InvalidRequest: 400 Client Error: Bad Request for url: http://substra-backend.node-1.com/data_manager/: [{'name': ['This field may not be null.']}]
```

**Comment by Fabien-GELUS** [comment]
I got the same error with this command : sudo python3 scripts/add_dataset_objective.py
```
Adding dataset...
Requests error status 400: {"message":[{"name":["This field may not be null."]}],"pkhash":"c1d9b42d9538f825109c38c4f599d47391347d198a0770331ba790b7ebcfaa40"}
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/substra/sdk/rest_client.py", line 126, in __request
    r.raise_for_status()
  File "/usr/lib/python3/dist-packages/requests/models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http://substra-backend.node-1.com/data_manager/

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "scripts/add_dataset_objective.py", line 95, in <module>
    dataset_key = client.add_dataset(DATASET, exist_ok=True)['pkhash']
  File "/usr/local/lib/python3.8/dist-packages/substra/sdk/client.py", line 41, in wrapper
    return f(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/substra/sdk/client.py", line 288, in add_dataset
    res = self._add(assets.DATASET, data, files=files, exist_ok=exist_ok)
  File "/usr/local/lib/python3.8/dist-packages/substra/sdk/client.py", line 160, in _add
    return self.client.add(
  File "/usr/local/lib/python3.8/dist-packages/substra/sdk/rest_client.py", line 261, in add
    return self._add(name, exist_ok=exist_ok, **request_kwargs)
  File "/usr/local/lib/python3.8/dist-packages/substra/sdk/rest_client.py", line 237, in _add
    return self.request('post', name, **request_kwargs)
  File "/usr/local/lib/python3.8/dist-packages/substra/sdk/utils.py", line 170, in wrapper
    return f(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/substra/sdk/rest_client.py", line 189, in request
    response = self._request(
  File "/usr/local/lib/python3.8/dist-packages/substra/sdk/rest_client.py", line 170, in _request
    return self.__request(request_name, url, **request_kwargs)
  File "/usr/local/lib/python3.8/dist-packages/substra/sdk/rest_client.py", line 138, in __request
    raise exceptions.InvalidRequest.from_request_exception(e)
substra.sdk.exceptions.InvalidRequest: 400 Client Error: Bad Request for url: http://substra-backend.node-1.com/data_manager/: [{'name': ['This field may not be null.']}]
```

**Comment by natct10** [comment]
> I get the same error with the latest commit of this branch with the command : sudo python3 scripts/add_dataset_objective.py
> 
> ```
> Adding dataset...
> Requests error status 400: {"message":[{"name":["This field may not be null."]}],"pkhash":"c1d9b42d9538f825109c38c4f599d47391347d198a0770331ba790b7ebcfaa40"}
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.8/dist-packages/substra/sdk/rest_client.py", line 126, in __request
>     r.raise_for_status()
>   File "/usr/lib/python3/dist-packages/requests/models.py", line 940, in raise_for_status
>     raise HTTPError(http_error_msg, response=self)
> requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http://substra-backend.node-1.com/data_manager/
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "scripts/add_dataset_objective.py", line 95, in <module>
>     dataset_key = client.add_dataset(DATASET, exist_ok=True)['pkhash']
>   File "/usr/local/lib/python3.8/dist-packages/substra/sdk/client.py", line 41, in wrapper
>     return f(*args, **kwargs)
>   File "/usr/local/lib/python3.8/dist-packages/substra/sdk/client.py", line 288, in add_dataset
>     res = self._add(assets.DATASET, data, files=files, exist_ok=exist_ok)
>   File "/usr/local/lib/python3.8/dist-packages/substra/sdk/client.py", line 160, in _add
>     return self.client.add(
>   File "/usr/local/lib/python3.8/dist-packages/substra/sdk/rest_client.py", line 261, in add
>     return self._add(name, exist_ok=exist_ok, **request_kwargs)
>   File "/usr/local/lib/python3.8/dist-packages/substra/sdk/rest_client.py", line 237, in _add
>     return self.request('post', name, **request_kwargs)
>   File "/usr/local/lib/python3.8/dist-packages/substra/sdk/utils.py", line 170, in wrapper
>     return f(*args, **kwargs)
>   File "/usr/local/lib/python3.8/dist-packages/substra/sdk/rest_client.py", line 189, in request
>     response = self._request(
>   File "/usr/local/lib/python3.8/dist-packages/substra/sdk/rest_client.py", line 170, in _request
>     return self.__request(request_name, url, **request_kwargs)
>   File "/usr/local/lib/python3.8/dist-packages/substra/sdk/rest_client.py", line 138, in __request
>     raise exceptions.InvalidRequest.from_request_exception(e)
> substra.sdk.exceptions.InvalidRequest: 400 Client Error: Bad Request for url: http://substra-backend.node-1.com/data_manager/: [{'name': ['This field may not be null.']}]
> ```

Thank you for this input @Fabien-GELUS !

File: examples/titanic/scripts/add_dataset_objective.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmorel) [reply examples/titanic/scripts/add_dataset_objective.py:81]: Reopening: the permissions here should also be included on line 143.

File: examples/titanic/scripts/add_train_algo_constant.py [hunk 2]
```diff
@@ -26,7 +26,11 @@
 
 ALGO = {
     'name': 'Constant death predictor',
-    'description': os.path.join(assets_directory, 'algo_constant/description.md')
+    'description': os.path.join(assets_directory, 'algo_constant/description.md'),
+    "permissions": {
+        "public": False,
+        "authorized_ids": []
+    },
```
Comment (jmorel) [inline examples/titanic/scripts/add_train_algo_constant.py:33]: You also need to include these permissions on line 66.

**Comment by natct10** [comment]
Thank you guys!

**Pull request merged**